src/main/java/com/google/sps/servlets/OverTimeCasesServlet.java

```diff
@@ -0,0 +1,286 @@
+      ArrayList<Integer> cases = new ArrayList<Integer>();
+      String territory = "";
+
+      // If the name of the territory contains a ", the offset needs to be bumped
+      int tempCoordOffset = coordOffset;
+      int tempTerritoryOffset = territoryOffset;
+      int tempDataOffset = dataOffset;
+      if (cells[0 + tempCoordOffset].contains("\"") || cells[1 + tempCoordOffset].contains("\"")) {
+        ++tempCoordOffset;
+        ++tempDataOffset;
+        ++tempTerritoryOffset;
+      }
+      // Ignore unassigned entires
+      if (cells[2 + tempCoordOffset].equals("")
+          || cells[3 + tempCoordOffset].equals("")) { // Entries represent coordinates
+        continue;
+      }
+      if (cells[2 + tempCoordOffset].equals("0.0")
+          || cells[3 + tempCoordOffset].equals("0.0")) { // No coodrinates
+        continue;
+      }
+
+      if (!cells[0 + tempTerritoryOffset].equals("")) { // Entry represents territory name
+        territory = cells[0 + tempTerritoryOffset];
+      } else if (!cells[1 + tempTerritoryOffset].equals(
+                     "")) { // For countires, name will appear in index 1
+        territory = cells[1 + tempTerritoryOffset];
+      }
+      // Coordinates appear in these indicies
+      double lat = Double.parseDouble(cells[2 + tempCoordOffset]);
+      double lng = Double.parseDouble(cells[3 + tempCoordOffset]);
+
+      // For building total worldwide history
+      if (firstAccess) {
+        firstAccess = false;
+        for (int i = 4 + tempDataOffset; i < cells.length; ++i) {
+          if (!cells[i].equals("")) {
+            int numCase = Integer.parseInt(cells[i]);
+            cases.add(numCase);
+            // If looking at global cases, create new entry in world array
+            if (coordOffset == 0) {
+              worldCases.add(numCase);
+            }
+          }
+        }
+      } else {
+        int globalOffset = -1;
+        for (int i = 4 + tempDataOffset; i < cells.length; ++i) {
+          if (!cells[i].equals("")) {
+            int numCase = Integer.parseInt(cells[i]);
+            cases.add(numCase);
+            // If looking at global cases, add to entry in world array
+            if (coordOffset == 0) {
+              ++globalOffset;
+              worldCases.set(globalOffset, worldCases.get(globalOffset) + numCase);
+            }
+          }
+        }
+      }
+      timeReports.put(new LocLatLng(territory, lat, lng), cases);
+    }
+
+    scanner.close();
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getRequestParameterOrDefault(
+      HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  /**
+   * Maintains location name with its coordinates. Used as key
+   */
+  class LocLatLng {
```

Comment:
more style (although im not 100% sure about this, more like 90-95), all classes should be defined as public or private, i think it defaults to public, but I think since you defined that this servlet as a public class, the inner classes should be defined too (private if they're only used by this class, and public if they're used by other classes